apps/src/dance/blockly/setup.ts

```diff
@@ -0,0 +1,28 @@
+import danceBlocks from '@cdo/apps/dance/blockly/blocks';
+import * as commonBlocks from '@cdo/apps/blocksCommon';
+import {LevelProperties} from '@cdo/apps/lab2/types';
+
+const blockUtils = require('@cdo/apps/block_utils');
+
+// Set up the global Blockly environment for Dance Party Lab2.
+export function setUpBlocklyForDanceLab(
+  levelProperties: LevelProperties | undefined
+) {
+  const blocksModule = danceBlocks;
+  const skin = levelProperties?.skin || undefined;
```

Comment:
These are used when `commonBlocks` [are installed](https://github.com/code-dot-org/code-dot-org/blob/1dd203194f6df9ed3802c8dcc2338f74af47ef06/apps/src/blocksCommon.js#L10). Common blocks are  installed in Dance. You can view them in the first screenshot above (BEFORE update).

Reply:
Revisiting this, I'm curious if we can remove the dependency on `levelProperties` here so that we only need to install Dance blocks once per script. A few thoughts
- We could break this up and have `commonBlocks.install()` in a separate function that depends on level properties, and remove the level properties dependency here? That way, this function can be called as a one-time setup function, while the `commonBlocks.install()` is the only thing that's called when level properties change. We can also use selectors specifically for `isK1` and `skin` so that we'd only reinstall common blocks if those specific values change, rather than whenever the entire levelProperties field changes (which would realistically happen on every level).
- Don't know the answer to this, but I'm curious if `isK1` and `skin` would actually change within a single script? We could also just make an assumption that we use the values from the first level we load. But not sure if that's necessarily a safe assumption, I wouldn't want us to get locked into this pattern if that changes.

Reply:
Interesting idea to install Dance blocks once per script - definitely open to trying it out!
Not sure about `isK1` and `skin` changing across a script - `K1` doesn't seem like it would change. I see that`skin` is assigned [here](https://github.com/code-dot-org/code-dot-org/blob/06574efbd8742ad0298ad5414f3ca9d5d90a8dfc/apps/src/appMain.js#L71) in `appMain.js` and is used to set the block field images in `blocksCommon`. I agree that neither seem like they would change across a script, but not sure.

Reply:
Recapping our discussion offline -
I completely missed that we still need `levelProperties` for accessing `sharedBlocks`. In that case I'd recommend we break this into two functions, one that just installs common blocks using `isK1` and `skin`, and another that installs shared blocks using `sharedBlocks`. That way when we call these functions in `DanceView`, each can be in their own `useEffect` and will only be called when those specific values change (instead of being called whenever the entire `levelProperties` object updates, which would be on every level).

Reply:
Just updated and confirmed locally that common blocks are NOT re-installed after level change.